lib/puppet/type/chocolateysource.rb

```diff
@@ -120,6 +111,10 @@ def insync?(is)
     if provider.respond_to?(:validate)
       provider.validate
     end
+
+    if (self[:ensure].to_sym == :present && (self[:location].nil? || self[:location] == ''))
+      raise ArgumentError, "A non-empty location must be specified when not disabling/removing a source."
```

Comment:
Not exactly a double negative. 

non-empty is an adjective describing location.

The second part is about when it must be specified, as in when it is ensure => present (or the default of that when it is not specified). It was becoming a lot of words to express the same thing more succinctly.